### ISSUE TITLE
"Non-exhaustive patterns" when debugging bytecode that contains SELFDESTRUCT

### ISSUE THREAD

**Issue by snd** [dapphub/dapptools#632]
`hevm` crashes with `Non-exhaustive patterns in Just this` when stepping through bytecode that contains `SELFDESTRUCT` in `--debug` mode.

### Minimal Example

```
$ hevm version
0.45.0
$ hevm exec --code 0x7f0000000000000000000000001230000000000000000000000000000000000456ff --gas 0xffffff --debug
```
press `n` twice to step through bytecode.
```
hevm: src/EVM/TTY.hs:854:5-34: Non-exhaustive patterns in Just this
```

The bytecode used above has been generated by assembling the following using `evmasm -a`:
```
PUSH32 0x1230000000000000000000000000000000000456
SELFDESTRUCT
```

**Comment by MrChico** [comment]
Hi! Thanks for reporting! Will take a look!

**Comment by MrChico** [comment]
fixed by #636
